### PR TITLE
Further support for custom block path types

### PR DESCRIPTION
**Pull request by bs2609**
Partial replacement of #4598, this just patches `WalkNodeProcessor.checkNeighborBlocks` to use the existing `getAiPathNodeType` function in addition to the current hardcoded block checks.

This means that blocks which return a "damage" path type will have surrounding blocks with the corresponding "danger" path type, as is currently the case for vanilla fire and cactus blocks.

This also replaces the need for the `isBurning` check here, as that's covered by the default `getAiPathNodeType` implementation.

### CODE REVIEW

**Comment by bs2609** [comment]
As a further addition, this adds an `EntityLiving` parameter to `getAiPathNodeType`, which allows blocks to return a different path type for different entities - e.g. for blocks which are only harmful to some entities.

Included here as it's pretty strongly related, but it can be split off if people want.

**Comment by liach** [comment]
if i recall correctly, entity livings already have their own override settings (not world/pos aware)

File: patches/minecraft/net/minecraft/pathfinding/FlyingNodeProcessor.java.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+         pathnodetype = this.func_193577_a(p_186319_1_, p_186319_2_, p_186319_3_, p_186319_4_, p_186319_6_, p_186319_7_, p_186319_8_, p_186319_9_, p_186319_10_, enumset, pathnodetype, blockpos);
++        this.currentEntity = null;
+ 
+         if (enumset.contains(PathNodeType.FENCE))
+         {
+@@ -321,6 +323,7 @@
+                 {
+                     pathnodetype = PathNodeType.DAMAGE_CACTUS;
+                 }
++                else if (pathnodetype1 == PathNodeType.DAMAGE_OTHER) pathnodetype = PathNodeType.DAMAGE_OTHER;
```
Comment (tterrag1098) [inline patches/minecraft/net/minecraft/pathfinding/FlyingNodeProcessor.java.patch:17]: What's the purpose of this change?

Comment (bs2609) [reply patches/minecraft/net/minecraft/pathfinding/FlyingNodeProcessor.java.patch:17]: This is the same as the equivalent change in `WalkNodeProcessor`, although the decompiling here arranges the logic slightly differently (and a little strangely).

The logic here considers positions to be "damage" if the position directly under them is "damage". The vanilla logic only applies this to "catus" and "fire" damage types, probably because the hardcoded logic in `getPathNodeTypeRaw` doesn't return a type of `DAMAGE_OTHER`. As this type can now be returned by blocks (since #3546), it should be handled correctly as well.

**Review by tterrag1098 — APPROVED** [review]

**Review by mezz — APPROVED** [review]
